fontforge/python.c

```diff
@@ -1735,50 +1735,100 @@ return( Py_BuildValue("i",ret));
 }
 
 static PyObject *PyFF_askChoices(PyObject *UNUSED(self), PyObject *args) {
-    char *title=NULL,*quest=NULL, **answers;
-    int def=0, cnt;
-    PyObject *answero;
-    int i, ret;
+    char *title=NULL, *quest=NULL;
+    char **answers; // receives answers written into `answero`
+    PyObject* defo = NULL; // default answer index, or tuple of len cnt
+    int def = 0; // default index, only used if not multianswer
+    int cnt; // number of answers
+    PyObject *multipleo; // expected to be boolean
+    bool multiple = false;
+    PyObject *answero; // expected to be tuple
+    int ret;
 
     if ( no_windowing_ui ) {
-	PyErr_Format(PyExc_EnvironmentError, "No user interface");
-return( NULL );
+        PyErr_Format(PyExc_EnvironmentError, "No user interface");
+        return( NULL );
     }
 
-    if ( !PyArg_ParseTuple(args,"esesO|i","UTF-8", &title, "UTF-8", &quest, &answero, &def) )
-return( NULL );
+    if ( !PyArg_ParseTuple(args,"ssO|OO", &title, &quest, &answero, &defo, &multipleo) )
+        return( NULL );
+
+    multiple = multipleo == Py_True;
+
     if ( !PySequence_Check(answero) || STRING_CHECK(answero)) {
-	PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+        PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
+        PyMem_Free(title);
+        PyMem_Free(quest);
+        return( NULL );
     }
+
     cnt = PySequence_Size(answero);
-    answers = calloc(cnt+1, sizeof(char *));
-    if ( def<0 || def>=cnt ) {
-	PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+    char* sel = calloc(cnt, sizeof(char));
+
+    if (defo != NULL && defo != Py_None)
+    if (multiple) {
```

Comment:
The default specification seems overly restrictive to me, in the sense that if you can accept multiple answers but only want to specify one default, it seems like you should be able to do that with an integer. 

That is, why can't the logic be: "if default is a tuple then`multiple` should be true, otherwise it should be an integer"?

Reply:
So, drop the `multiple` argument and infer it from the type of `default`? There are cases where you'd want default to be None but multiple selectable choices though.

Reply:
No, leave `multiple` and continue to reject passing a tuple for the defaults when `multiple` isn't true. But allow a caller to specify a single default with an integer when `multiple` is true if they feel like it. Just treat that as equivalent to passing the boolean tuple with one true value. 

This is tricky because that's definitely the wrong way to treat the return value -- when `multiple` is true you always want to return a tuple. Otherwise the caller would have to deal with annoying special cases when the user selects a single value. But input parameters are up to the caller and it's better not to force the author to build a tuple structure if they don't need to.

Reply:
Understood, thanks :) will do it when I get home from church